### PR TITLE
Added Mercury, Soyuz and Soyuz XL Macropads

### PR DESCRIPTION
**Pull request by spacehangover**
Added Mercury, Soyuz and Soyuz XL Macropads by WWA

Added 3 new macropads by WWA

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/wwa/soyuz/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "analog.h"
+#include "qmk_midi.h"
+
+enum custom_keycodes { 
+    DEFAULT= SAFE_RANGE,SPOTIFY,DISCORD,CHROME/*,VLC*/
+};
+
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuz/keymaps/default/keymap.c:11]: Unused.
```suggestion
```

File: keyboards/handwired/wwa/soyuz/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    if (abs(currentPotValue3 - previousPotValue3) > threshold) {
+        midi_send_cc(&midi_device, 4, 0x0C, 0x7F - (currentPotValue3 >> 3));
+        previousPotValue3 = currentPotValue3;
+    }
+    
+}
+
+void matrix_scan_user(void) {
+    slider();
+}
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuz/keymaps/default/keymap.c:58]: This should be moved to the keyboard level in a `soyuz.c` using the appropriate `_kb` functions.
```suggestion
```

File: keyboards/handwired/wwa/soyuz/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "analog.h"
+#include "qmk_midi.h"
+
+enum custom_keycodes { /* In order to add more volume controls, add here a custom keycode, ex: VLC */
+    DEFAULT= SAFE_RANGE,SPOTIFY,DISCORD,CHROME/*,VLC*/
+};
+
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuz/keymaps/via/keymap.c:11]: Unused.
```suggestion
```

File: keyboards/handwired/wwa/soyuz/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    if (abs(currentPotValue3 - previousPotValue3) > threshold) {
+        midi_send_cc(&midi_device, 4, 0x0C, 0x7F - (currentPotValue3 >> 3));
+        previousPotValue3 = currentPotValue3;
+    }
+    
+}
+
+void matrix_scan_user(void) {
+    slider();
+}
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuz/keymaps/via/keymap.c:57]: As above.
```suggestion
```

File: keyboards/handwired/wwa/soyuz/readme.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# soyuz
+
+![soyuz](https://imgur.com/a/sOYdpzS)
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuz/readme.md:3]:
```suggestion
![soyuz](https://i.imgur.com/KwwORGSh.jpeg)
```

File: keyboards/handwired/wwa/soyuz/rules.mk [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+MIDI_ENABLE = yes           # MIDI support
+AUDIO_ENABLE = no           # Audio output
+LTO_ENABLE = yes
+
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuz/rules.mk:11]:
```suggestion
```

File: keyboards/handwired/wwa/soyuzxl/config.h [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl/config.h:31]:
```suggestion
```

File: keyboards/handwired/wwa/soyuzxl/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    if (abs(currentPotValue5 - previousPotValue5) > threshold) {
+        midi_send_cc(&midi_device, 6, 0x0E, 0x7F - (currentPotValue5 >> 3));
+        previousPotValue5 = currentPotValue5;
+    }
+    
+}
+
+void matrix_scan_user(void) {
+    slider();
+}
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl/keymaps/default/keymap.c:76]: As above.
```suggestion
```

File: keyboards/handwired/wwa/soyuzxl/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "analog.h"
+#include "qmk_midi.h"
+
+enum custom_keycodes { 
+    DEFAULT= SAFE_RANGE,SPOTIFY,DISCORD,CHROME/*,VLC*/
+};
+
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl/keymaps/default/keymap.c:11]:
```suggestion
```

File: keyboards/handwired/wwa/soyuzxl/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "analog.h"
+#include "qmk_midi.h"
+
+enum custom_keycodes { 
+    DEFAULT= SAFE_RANGE,SPOTIFY,DISCORD,CHROME/*,VLC*/
+};
+
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl/keymaps/via/keymap.c:11]:
```suggestion
```

File: keyboards/handwired/wwa/soyuzxl/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    if (abs(currentPotValue5 - previousPotValue5) > threshold) {
+        midi_send_cc(&midi_device, 6, 0x0E, 0x7F - (currentPotValue5 >> 3));
+        previousPotValue5 = currentPotValue5;
+    }
+    
+}
+
+void matrix_scan_user(void) {
+    slider();
+}
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl/keymaps/via/keymap.c:76]: As above.
```suggestion
```

File: keyboards/handwired/wwa/soyuzxl/readme.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
+# soyuzxl
+
+![soyuzxl](https://imgur.com/gallery/AqPRXY7)
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl/readme.md:3]:
```suggestion
![soyuzxl](https://i.imgur.com/njUq2s3h.jpeg)
```

File: keyboards/handwired/wwa/soyuzxl/rules.mk [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+MIDI_ENABLE = yes           # MIDI support
+AUDIO_ENABLE = no           # Audio output
+LTO_ENABLE = yes
+
```
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl/rules.mk:11]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by spacehangover** [comment]
Hey, thanks for the review, will make the changes as requested, i just have one question, what do you mean by this? This should be moved to the keyboard level in a soyuz.c using the appropriate _kb functions. 
Referring to the MIDI part of my code, i wasn't able to find any docs regarding the use of MIDI, so i copied the code from another keyboard already merged in QMK.

**Comment by tzarc** [comment]
This should be three separate PRs, one per keyboard.

**Pull request closed without merging**